Week3/app.js

```diff
@@ -0,0 +1,278 @@
+const util = require('util');
+const mysql = require('mysql');
+//const uuidv4 = require('uuid-v4');//new id
+
+const connection = mysql.createConnection({
+  host: 'localhost',
+  user: 'hyfuser',
+  password: 'hyfpassword',
+  database: 'ToDoDB'
+});
+
+const execQuery = util.promisify(connection.query.bind(connection));
+let activeUserId = 1;
+
+/**Category */
+async function listCategory() {
+  let queryWhere = " where id in (select categoryId from userscatalogs where userid = " + activeUserId + ")";
+  const categList = await getResult('categories', queryWhere);
+  return '' + JSON.stringify(categList);
+}
+async function createCategory(name) {
+  //uuidv4()
+  const id = await getLastId('categories');
+  let query = "insert into categories values (" + id + ", '" + name + "')";
+  await executeQuery(query);
+  //When category is created, relation of user & category is also created.
+  await createUsersCatalogs(activeUserId, id, true, true);
+  return name + ' ==> New category is created.';
+}
+async function updateCategory(id, newName) {
+  let query = "update categories set name = '" + newName + "' where id=" + id;
```

Comment:
Just a hint, try to escape this values ```newName``` and ```id``` so you won't need to worry about SQL Injection. =)